Adjust README.md to changes in the API (#451)
golemfactory/yapapi: README.md

  read files) with it. This image needs to be packed and uploaded into Golem's image repository
  using our dedicated tool - [`gvmkit-build`](https://pypi.org/project/gvmkit-build/).
* **create your requestor agent** - this is where `yapapi` comes in. Utilizing our high-level
  API, the creation of a requestor agent should be straightforward and require minimal effort.
  You can use examples contained in this repository as references, the directory
  [`examples/hello-world/`](https://github.com/golemfactory/yapapi/tree/master/examples/hello-world)
  contains minimal examples of fully functional requestor agents and
  is therefore the best place to start exploring.

### Components

There are a few components that are crucial for any requestor agent app:

#### Golem

The heart of the high-level API is the `Golem` class (`yapapi.Golem`), which serves as 
the "engine" of a requestor agent. `Golem` is responsible for finding providers interested in
the jobs you want to execute, negotiating agreements with them and processing payments. It also

> implements core functionality required for sending commands to providers that 
> agreed to run your tasks.

Reviewer: ```suggestion
implements core functionality required to execute commands on providers that have signed such agreements.
```

`Golem` provides two entry points for executing jobs on the Golem network, corresponding to the
two basic modes of operation of a requestor agent:

* The method `execute_tasks` allows you to submit a _task-based_ job for execution. Arguments
to this method must include a sequence of independent _tasks_ (units of work) to be distributed
among providers, a _payload_ (a VM image) required to compute them, and a _worker_ function, which
will be used to convert each task to a sequence of steps to be executed on a provider. You may also
specify the timeout for the whole job, the maximum number of providers used at any given time,
and the maximum amount that you want to spend.

* The method `run_service` allows you, as you probably guessed, to run a _service_ on Golem. 
Instead of a task-processing worker function, an argument to `run_service` is a
class (a subclass of `yapapi.Service`) that implements the behaviour of your service in various
stages of its lifecycle (when it's _starting_, _running_ etc.). Additionally, you may specify
the number of _service instances_ you want to run and the service expiration datetime.

> Prior to version `0.6.0`, only task-based jobs could be executed. They are probably conceptually
> simpler than services, and therefore in this `README` we focus on jobs of this type. 
> For more information on both types of jobs please refer to our [handbook](https://handbook.golem.network).

Reviewer: seems redundant for me... maybe we should a short intro to Services instead? (in a separate PR perhaps?)

```suggestion
Prior to version `0.6.0`, only task-based jobs could be executed. For more information on both types of jobs please refer to our [handbook](https://handbook.golem.network).
```

the `worker` parameter) to the `execute_tasks` method of `Golem`.

The worker receives a _work context_ (`yapapi.WorkContext`) object that serves
as an interface between your script and the execution unit within the provider.
Using the work context, you define the steps that the provider needs to execute in order
to complete the job you're giving them - e.g. transferring files to and from the provider
or running commands within the execution unit on the provider's end.

Depending on the number of workers, and thus, the maximum number of providers that 
`execute_tasks` utilizes in parallel, a single worker may tackle several tasks
and you can differentiate the steps that need to happen once
per worker run, which usually means once per provider node - but that depends on the
exact implementation of your worker function - from those that happen for each
task. An example of the former would be an upload of a source
file that's common to each task; and of the latter - a step that triggers the
processing of the file using a set of parameters specified for a particular task.

#### Task

The _task_ (`yapapi.Task`) object describes a unit of work that your application needs
to carry out.

> A `Golem` object will feed an instance of your worker - bound to a single provider node -

Reviewer: ```suggestion
`Golem` will feed an instance of your worker - bound to a single provider node -
```

with `Task` objects. The worker will be responsible for completing those tasks. Typically,
it will turn each task into a sequence of steps to be executed in a single run
of the execution script on a provider's machine, in order to compute the task's result.

### Example

> An example task-based Golem application, using a Docker image containing Alpine Linux installation

Reviewer: I'd abstract from specifying such details here

```suggestion
An example task-based Golem application, using a minimal Docker image
```
